src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/probation/CommunityApiClient.kt

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+
+@Component
+class CommunityApiClient(@Qualifier("oauthCommunityApiClient") val communityApiClient: WebClient) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getTeamsCodesForUser(staffIdentifier: Long): List<String> {
+    try {
+      val user = communityApiClient
+        .get()
+        .uri("/secure/staff/staffIdentifier/$staffIdentifier")
+        .accept(MediaType.APPLICATION_JSON)
+        .retrieve()
+        .bodyToMono(User::class.java)
+        .block()
+      if (user === null) {
```

Comment:
this should never be null and if it was then throwing an exception would be a fair enough response rather than failing silently. 

So few options:  

could `!!` - you'd get a null pointer referencing that line of code but not a lot of context in the message 

could ` ?: throw IllegalStateException("some context")